### PR TITLE
[FIX] SOM: Fix crash when color is constant

### LINKED ISSUE #5858: SOM: IndexError when setting color with constant variable
**What's wrong?**

SOM crashes with an IndexError when selecting a variable with constant values in Color combo box

**How can we reproduce the problem?** 

Run the following test case:
```
    def test_colored_circles_with_constant(self):
        with self.iris.unlocked():
            self.iris.X[:, 0] = 1
        self.send_signal(self.widget.Inputs.data, self.iris)
        simulate.combobox_activate_index(self.widget.controls.attr_color, 4)
```

**What's your environment?**

- Operating system: OSX
- Orange version: master
- How you installed Orange: pip

### PR DESCRIPTION
**Pull request by janezd**
##### Issue

Fixes #5858.

If the variable determining the color was constant, the binning set two thresholds, which were both removed, thus creating an empty list of thresholds and labels. This created problems later in the code.

##### Description of changes

Set a threshold and label.

##### Includes
- [X] Code changes
- [X] Tests

### CODE REVIEW

**Comment by VesnaT** [comment]
The fix works, except when the constant is NaN:

image

File: Orange/widgets/unsupervised/owsom.py [hunk 1]
```diff
@@ -210,6 +211,8 @@ class Warning(OWWidget.Warning):
         ignoring_disc_variables = Msg("SOM ignores categorical variables.")
         missing_colors = \
             Msg("Some data instances have undefined value of '{}'.")
+        no_defined_colors = \
+            Msg("'{}' has node defined values.")
```
Comment (VesnaT) [inline Orange/widgets/unsupervised/owsom.py:215]: A typo?

Comment (janezd) [reply Orange/widgets/unsupervised/owsom.py]: yes

**Comment by VesnaT** [comment]
Similarly, the plot can't be colored by a Time Variable.

image

**Comment by janezd** [comment]
> Similarly, the plot can't be colored by a Time Variable.

This problem was actually in `time_binning` and would happen in any widget using this for discretization.

Which, by the way, reminds me: all widget should use `time_binning`/`decimal_binning` for discretization. E.g. Mosaic and Sieve, as well as all widgets that show color legends.

**Pull request merged**